src/components/advanced-color-picker/advanced-color-picker.component.tsx

```diff
@@ -11,66 +10,122 @@ import { SimpleColorPicker, SimpleColor } from "../simple-color-picker";
+  name: string;
+  /** Prop for `onBlur` event */
+  onBlur?: (ev: React.FocusEvent<HTMLInputElement>) => void;
+  /** Prop for `onChange` event */
+  onChange?: (ev: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Prop for `onClose` event */
+  onClose?: (ev: React.MouseEvent<HTMLElement>) => void;
+  /** Prop for `onOpen` event */
+  onOpen?: (ev: React.MouseEvent<HTMLElement>) => void;
+  /** Prop for `open` status */
+  open?: boolean;
+  /** The ARIA role to be applied to the component container */
+  role?: string;
+  /** Prop for `selectedColor` containing pre-selected color for `controlled` use */
+  selectedColor?: string;
+}
 
 const AdvancedColorPicker = ({
   "aria-describedby": ariaDescribedBy,
   "aria-label": ariaLabel,
   "aria-labelledby": ariaLabelledBy,
+  availableColors,
+  defaultColor,
   name,
-  open,
   onOpen,
   onClose,
   onChange,
   onBlur,
-  availableColors,
-  defaultColor,
-  selectedColor,
+  open,
   role,
+  selectedColor,
   ...props
-}) => {
+}: AdvancedColorPickerPropTypes) => {
   const isOpen = open || false;
-  const [dialogOpen, setDialogOpen] = useState();
+  const [colors, setColors] = useState<
+    { value: string; label: string; getRef: () => HTMLInputElement | null }[]
+  >();
+  const [dialogOpen, setDialogOpen] = useState<boolean>();
   const currentColor = selectedColor || defaultColor;
-  const [selectedColorRef, setSelectedColorRef] = useState();
-
-  const gridItemRefs = useRef(
-    Array.from(
-      {
-        length: availableColors.length,
-      },
-      () => React.createRef()
-    )
+  const [
+    selectedColorRef,
+    setSelectedColorRef,
+  ] = useState<HTMLInputElement | null>(null);
+
+  const simpleColorPickerData = useRef<{
+    gridItemRefs: Array<HTMLInputElement | null>;
+  }>();
+
+  useEffect(
+    () =>
+      setColors(
+        availableColors.map(({ value, label }, index) => {
+          return {
+            value,
+            label,
+            getRef: () =>
+              /* istanbul ignore next */
+              simpleColorPickerData.current
+                ? simpleColorPickerData.current.gridItemRefs[index]
+                : null,
+          };
+        })
+      ),
+    [availableColors]
   );
 
-  const colors = availableColors.map(({ value, label }, index) => {
-    return {
-      value,
-      label,
-      ref: gridItemRefs.current[index],
-    };
-  });
-
   useEffect(() => {
     if (dialogOpen || isOpen) {
-      const selected = colors.find((c) => currentColor === c.value);
-      setSelectedColorRef(selected.ref.current);
+      const newColor = colors?.find((c) => currentColor === c.value);
+
+      /* istanbul ignore else */
+      if (newColor) {
+        setSelectedColorRef(newColor.getRef());
+      }
     }
   }, [colors, currentColor, dialogOpen, isOpen]);
 
   const handleFocus = useCallback(
     (e, firstFocusableElement) => {
+      /* istanbul ignore else */
```

Comment:
we do need to check if the key is `Tab`, nothing more. If you simplify that it would just look like:
```
if (e.key === "Tab") {
  .
  .
  .
}
```
so I can't see what else should we cover here.

This was covered by an unit test in which we press other key than tab, but I see no sense in having a test like that.